src/globus_sdk/services/gcs/errors.py

```diff
@@ -26,10 +26,10 @@ def _get_args(self) -> list[t.Any]:
             args.append(self.detail)
         return args
 
-    def _load_from_json(self, data: dict[str, t.Any]) -> None:
-        super()._load_from_json(data)
+    def _final_parse_callback(self) -> bool:
```

Comment:
I like the idea of this extension for polymorphism a lot.

---

Loosely held naming opinion, if you disagree feel free to simply ignore

Seems very hook-y so I'd consider naming it with "post" over "final" (sets us up to add a "pre" if needbe).

```suggestion
    def _post_parse_hook(self) -> bool:
```